src/backend/egl/error.rs

```diff
@@ -153,7 +153,28 @@ impl EGLError {
 }
 
 /// Wraps a raw egl call and returns error codes from `eglGetError`, if it fails.
-pub fn wrap_egl_call<R, F: FnOnce() -> R>(call: F) -> Result<R, EGLError> {
+pub fn wrap_egl_call<R: PartialEq, F: FnOnce() -> R>(call: F, err: R) -> Result<R, EGLError> {
     let res = call();
-    EGLError::from_last_call().map(|()| res)
+    if res != err {
+        Ok(res)
+    } else {
+        EGLError::from_last_call().map(|()| res)
+    }
+}
+
+/// Wraps a raw egl call and returns error codes from `eglGetError`, if it fails.
+pub fn wrap_egl_call_ptr<R, F: FnOnce() -> *const R>(call: F) -> Result<*const R, EGLError> {
```

Comment:
Same here, maybe just a short description that the pointer returned from
the wrapped call is checked against `null`